internal/controllers/terraformlayer/conditions.go

```diff
@@ -243,6 +244,31 @@ func (r *Reconciler) IsApplyUpToDate(t *configv1alpha1.TerraformLayer) (metav1.C
 	return condition, true
 }
 
+func (r *Reconciler) IsSyncScheduled(t *configv1alpha1.TerraformLayer) (metav1.Condition, bool) {
+	condition := metav1.Condition{
+		Type:               "IsSyncScheduled",
+		ObservedGeneration: t.GetObjectMeta().GetGeneration(),
+		Status:             metav1.ConditionUnknown,
+		LastTransitionTime: metav1.NewTime(time.Now()),
+	}
+	// check if annotations.SyncNow is present
+	if _, ok := t.Annotations[annotations.SyncNow]; ok {
+		condition.Reason = "SyncScheduled"
+		condition.Message = "A sync has been manually scheduled"
+		condition.Status = metav1.ConditionTrue
+		// Remove the annotation to avoid running the sync again
+		err := annotations.Remove(context.Background(), r.Client, t, annotations.SyncNow)
+		if err != nil {
+			log.Errorf("Failed to remove annotation %s from layer %s: %s", annotations.SyncNow, t.Name, err)
+		}
```

Comment:
Removing the annotation right away is causing another reconciliation right?
Does the layer end in a `NoSyncScheduled` status just a few seconds later? Does it impact the Server API in its `SyncLayerHandler` ?

Reply:
The layer controller creates a run, and the run controller locks the layer related to a this new run until the run succeeds or fails. Since the layer is locked during the whole duration of the run, this should not cause any issue.